zingolib/src/lightclient/propose.rs

```diff
@@ -211,9 +219,26 @@ impl LightClient {
     /// Unstable function to expose the zip317 interface for development
     pub async fn propose_shield(&self) -> Result<ShieldProposal, ProposeShieldError> {
         let proposal = self.create_shield_proposal().await?;
-        self.store_proposal(ZingoProposal::Shield(proposal.clone()))
-            .await;
-        Ok(proposal)
+
+        // dont dust ourselves. we have to give ourselves a note with worthwhile value
+
+        let biggest_change = proposal
+            .steps()
+            .first()
+            .balance()
+            .proposed_change()
+            .iter()
+            .fold(0, |biggest_change_value_value, change_value| {
+                std::cmp::max(biggest_change_value_value, change_value.value().into_u64())
+            });
+
+        if biggest_change > SHIELDING_CUTOFF {
+            self.store_proposal(ZingoProposal::Shield(proposal.clone()))
+                .await;
+            Ok(proposal)
+        } else {
+            Err(ProposeShieldError::BelowThreshold(biggest_change))
+        }
```

Comment:
none of this makes sense to me. due to how @AloeareV constructed the `value_to_shield` in the ShieldCommand It is my understanding that fees for shielding can't be paid for from the transparent pool. therefore, if we *were* to check the value we are shielding we should do it the same way we do in commands.rs, by summing the transparent inputs. However ..... we have a shielding threshold on `propose_shielding` which means we don't shield dust anyway .. so this is a no-op right?

Reply:
i have just found that `propose_shield` also calls the same `compute_balance` so if we want to prevent dust change we can just rely on the `DustAction::Reject` policy. this doesnt need re-implementing

Reply:
> none of this makes sense to me. due to how @AloeareV constructed the `value_to_shield` in the ShieldCommand It is my understanding that fees for shielding can't be paid for from the transparent pool. therefore, if we _were_ to check the value we are shielding we should do it the same way we do in commands.rs, by summing the transparent inputs. However ..... we have a shielding threshold on `propose_shielding` which means we don't shield dust anyway .. so this is a no-op right?

looking at the shield code in LRZ my assumptions were wrong and the shielding amount in commands.rs must also be incorrect. There are no shielded inputs added in a shielding proposal and therefore all fees must be covered by transparent inputs.